hudi-hadoop-common/src/test/java/org/apache/parquet/avro/TestHoodieAvroReadSupport.java

```diff
@@ -0,0 +1,83 @@
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.parquet.avro;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.parquet.schema.ConversionPatterns;
+import org.apache.parquet.schema.GroupType;
+import org.apache.parquet.schema.MessageType;
+import org.apache.parquet.schema.OriginalType;
+import org.apache.parquet.schema.PrimitiveType;
+import org.apache.parquet.schema.Type;
+import org.apache.parquet.schema.Types;
+import org.junit.jupiter.api.Test;
+
+import java.util.Collections;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+class TestHoodieAvroReadSupport {
+  private final Type legacyListType = ConversionPatterns.listType(Type.Repetition.REQUIRED, "legacyList",
+      Types.primitive(PrimitiveType.PrimitiveTypeName.DOUBLE, Type.Repetition.REPEATED).named("double_field"));
+  private final Type listType = ConversionPatterns.listOfElements(Type.Repetition.REQUIRED, "newList",
+      Types.primitive(PrimitiveType.PrimitiveTypeName.DOUBLE, Type.Repetition.REPEATED).named("element"));
+  private final Type integerField = Types.primitive(PrimitiveType.PrimitiveTypeName.INT32, Type.Repetition.REQUIRED).named("int_field");
+  private final Type legacyMapType = new GroupType(Type.Repetition.OPTIONAL, "my_map", OriginalType.MAP, new GroupType(Type.Repetition.REPEATED, "map",
+      Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED).named("key"),
+      Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED).named("value")));
+  private final Type mapType = ConversionPatterns.stringKeyMapType(Type.Repetition.OPTIONAL, "newMap",
+      Types.primitive(PrimitiveType.PrimitiveTypeName.BINARY, Type.Repetition.REQUIRED).named("value"));
+  private final Configuration configuration = mock(Configuration.class);
+
+  @Test
+  void fileContainsLegacyList() {
+    when(configuration.getBoolean(AvroWriteSupport.WRITE_OLD_LIST_STRUCTURE, AvroWriteSupport.WRITE_OLD_LIST_STRUCTURE_DEFAULT)).thenReturn(false);
+    MessageType messageType = new MessageType("LegacyList", integerField, legacyListType, mapType);
+    new HoodieAvroReadSupport<>().init(configuration, Collections.emptyMap(), messageType);
+    verify(configuration).set(eq(AvroWriteSupport.WRITE_OLD_LIST_STRUCTURE), eq("true"), anyString());
+  }
+
+  @Test
+  void fileContainsLegacyMap() {
```

Comment:
do we also test nested array, nested map and nested map with value type as a list.

Reply:
Added new cases